### PR TITLE
Master phoenix 2 nby

### PR DESCRIPTION
**Pull request by Goaman**

### CODE REVIEW

File: addons/html_editor/static/src/editor/core/movenode_plugin.js [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+        node.parentElement?.getAttribute("contentEditable") === "true" &&
+        !node.parentElement.closest(".o_editor_banner")
+    );
+}
+
+function isPointInside(rect, x, y) {
+    return rect.left <= x && rect.right >= x && rect.top <= y && rect.bottom >= y;
+}
+
+const simpleDraggableHook = {
```
Comment (cammarosano) [inline addons/html_editor/static/src/editor/core/movenode_plugin.js]: Why do you prepend this variable name with "simple"? Are there any other draggable hooks?

File: addons/html_editor/static/src/editor/core/movenode_plugin.js [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+
+const ALLOWED_ELEMENTS =
+    "h1, h2, h3, p, hr, pre, blockquote, ul, ol, table, .o_knowledge_behavior_anchor, .o_text_columns, .o_editor_banner, .oe_movable";
+
+class MoveNode extends Plugin {
+    static name = "movenode";
+    static dependencies = ["selection", "container"];
+
+    setup() {
+        this._intersectionObserver = new IntersectionObserver(
```
Comment (cammarosano) [inline addons/html_editor/static/src/editor/core/movenode_plugin.js]: properties and methods prefixed with "_" ?

File: addons/html_editor/static/src/editor/core/container_plugin.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    ];
+
+    setup() {
+        // This main container is used to contain a tree of sub containers.
+        // By having one parent that contains a tree of containers, it is easy
+        // to change the z-index of any container by changing their place in the
+        // tree rather than tweaking a z-index number.
+        this.mainAbsoluteContainer = this.document.createElement("div");
+        this.mainAbsoluteContainer.classList.add("oe-absolute-container");
+        this.editable.before(this.mainAbsoluteContainer);
```
Comment (cammarosano) [inline addons/html_editor/static/src/editor/core/container_plugin.js]: Shouldn't you remove it on destroy?

File: addons/html_editor/static/src/editor/core/movenode_plugin.js (hunk 1 above)
Comment (Goaman) [reply addons/html_editor/static/src/editor/core/movenode_plugin.js]: these lines were not created by me originally

File: addons/html_editor/static/src/editor/core/container_plugin.js (hunk 3 above)
Comment (Goaman) [reply addons/html_editor/static/src/editor/core/container_plugin.js]: yep, you're right

File: addons/html_editor/static/src/main/movenode_plugin.js [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+            characterData: true,
+            characterDataOldValue: true,
+        });
+        this.addDomListener(window, "resize", this.updateHooks);
+        if (this.document.defaultView !== window) {
+            this.addDomListener(this.document.defaultView, "resize", this.updateHooks);
+        }
+    }
+    destroy() {
+        this.intersectionObserver.disconnect();
```
Comment (ged-odoo) [inline addons/html_editor/static/src/main/movenode_plugin.js:85]: super.destroy

**Pull request merged**